scaler/io/ymq/epoll_context.cpp

```diff
@@ -1,77 +1,73 @@
 #include "scaler/io/ymq/epoll_context.h"
-#include <format>
-#include "scaler/io/ymq/common.h"
 
-void EpollContext::registerEventManager(EventManager& em) {
-    epoll_event ev {
-        .events = EPOLLOUT | EPOLLIN | EPOLLET,  // Edge-triggered
-        .data   = {.ptr = &em},
-    };
+#include <sys/epoll.h>
 
-    epoll_fd.epoll_ctl(EPOLL_CTL_ADD, em._fd, &ev);
-}
+#include <cerrno>
+#include <functional>
 
-void EpollContext::removeEventManager(EventManager& em) {
-    epoll_fd.epoll_ctl(EPOLL_CTL_DEL, em._fd, nullptr);
-}
+#include "scaler/io/ymq/event_manager.h"
 
-void EpollContext::executePendingFunctors() {}
+namespace scaler {
+namespace ymq {
 
-void EpollContext::loop() {
-    std::array<epoll_event, 1024> events;
-    auto result = epoll_fd.epoll_wait(events.data(), 1024, -1);
+void EpollContext::execPendingFunctions() {
+    while (_delayedFunctions.size()) {
+        auto top = std::move(_delayedFunctions.front());
+        top();
+        _delayedFunctions.pop();
```

Comment:
I think `std::move()` might be unnecessary here.

```suggestion
        const DelayedFunction& top = _delayedFunctions.front();
        top();
        _delayedFunctions.pop();
```